### PR TITLE
Fixes #16974: Full review of rudder-lang suported systems

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/16974

### CODE REVIEW

**Comment by peckpeck** [comment]
Do not merge before we discuss and solve all the TODO in this file

**Comment by peckpeck** [comment]
Commit modified

File: rudder-lang/libs/oslib.rl [hunk 1]
```diff
@@ -39,16 +65,17 @@ items in bsd {
 }
 
 items in linux {
-  alpine_linux,
-  arch_linux,
+# TODO do we keep android ?
```
Comment (amousset) [inline rudder-lang/libs/oslib.rl:68]: I don't think it really matters as it is not a list of supported OS. I'm fine with both.

**Comment by amousset** [comment]
We probably need a SUSE fmaily with SLES and OpenSUSE

File: rudder-lang/libs/oslib.rl [hunk 2]
```diff
@@ -88,13 +111,275 @@ items in solaris {
...
+items in redhat_8 {
+  redhat_8_0,
+  redhat_8_1,
+  *
+}
+
+# == redhat: cfengine + wikipedia checked
+
+# == ubuntu
+# TODO Should we have major only for ubuntu ?
```
Comment (amousset) [inline rudder-lang/libs/oslib.rl:258]: 18_04 is not major+minor but a major version by itself. There is no particular link between 18_04 and 18_10, so I don't think so.

File: rudder-lang/libs/oslib.rl [hunk 3]
```diff
@@ -30,6 +30,32 @@ items in unix {
...
 
+# == AIX
+items in aix {
+  aix_5,
+  aix_6,
+  aix_7,
+  *
+}
+
+# TODO should we make it extendable ?
```
Comment (peckpeck) [inline rudder-lang/libs/oslib.rl:41]: The general question, is for all OS, should we add a * (extensible list) in os subversion, given that there won't be anymore such version. 
I think we should not have it

File: rudder-lang/libs/oslib.rl (hunk 1 above)
Comment (peckpeck) [reply rudder-lang/libs/oslib.rl:68]: let's go with a no, since this list is extensible

File: rudder-lang/libs/oslib.rl (hunk 2 above)
Comment (peckpeck) [reply rudder-lang/libs/oslib.rl:258]: I agree

File: rudder-lang/libs/oslib.rl [hunk 4]
```diff
@@ -88,13 +111,275 @@ items in solaris {
...
 }
+
+enum alias lenny    = debian_5
+enum alias squeeze  = debian_6
+enum alias wheezy   = debian_7
+enum alias jessie   = debian_8
+enum alias stretch  = debian_9
+enum alias buster   = debian_10
+enum alias bullseye = debian_11
+# TODO cfengine doesn't have proper minor version for debian 5 and 6
```
Comment (peckpeck) [inline rudder-lang/libs/oslib.rl:328]: The question is, if we add them here, how can we implement them ?

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
PR updated with a new commit
I think it's mergreable now

**Comment by peckpeck** [comment]
PR updated with a new commit

**Pull request closed without merging**